core/ide/src/main/java/org/alice/stageide/sceneeditor/StorytellingSceneEditor.java

```diff
@@ -604,112 +488,30 @@ private boolean isSelectableType(AbstractType<?, ?, ?> valueType) {
-    frontTransform.orientation.setValue(new ForwardAndUpGuide(Vector3.accessPositiveZAxis(), Vector3.accessPositiveYAxis()));
-    assert frontTransform.orientation.isWithinReasonableEpsilonOfUnitLengthSquared();
-    this.frontOrthoMarkerImp.setLocalTransformation(frontTransform);
-    picturePlane.setHeight(4);
-    this.frontOrthoMarkerImp.setPicturePlane(picturePlane);
-    orthographicCameraMarkerImps.add(this.frontOrthoMarkerImp);
-
-    this.startingCameraMarkerImp.getAbstraction().setName(MarkerUtilities.getNameForCameraImp(this.startingCameraMarkerImp));
-    this.layoutSceneMarkerImp.getAbstraction().setName(MarkerUtilities.getNameForCameraImp(this.layoutSceneMarkerImp));
-    this.topOrthoMarkerImp.getAbstraction().setName(MarkerUtilities.getNameForCameraImp(this.topOrthoMarkerImp));
-    this.sideOrthoMarkerImp.getAbstraction().setName(MarkerUtilities.getNameForCameraImp(this.sideOrthoMarkerImp));
-    this.frontOrthoMarkerImp.getAbstraction().setName(MarkerUtilities.getNameForCameraImp(this.frontOrthoMarkerImp));
-  }
-
   public Boolean isVrActive() {
     return isVrScene;
   }
 
   private void setIsVrActive(boolean isActive) {
     isVrScene = isActive;
-    startingCameraMarkerImp.setVrActive(isVrScene);
-    layoutSceneMarkerImp.setVrActive(isVrScene);
-
-    // Update menu icon
-    MarkerUtilities.addIconForCameraImp(startingCameraMarkerImp, isVrScene ? "vrHeadset" : "mainCamera");
+    mainCameraViewTracker.setIsVrActive(isActive);
 
     // Update camera marker move buttons
     MoveActiveCameraToMarkerActionOperation.getInstance().updateDisplay();
     MoveMarkerToActiveCameraActionOperation.getInstance().updateDisplay();
   }
 
-  private void clearCameras() {
-    this.snapGrid.stopTrackingCameras();
-    if (this.onscreenRenderTarget.getSgCameraCount() > 0) {
-      onscreenRenderTarget.clearSgCameras();
-    }
-    this.mainCameraViewTracker.setCameras(null, null);
-    this.globalDragAdapter.clearCameraViews();
-  }
-
-  private void setCameras(SymmetricPerspectiveCamera perspectiveCamera, OrthographicCamera orthographicCamera) {
-    this.globalDragAdapter.addCameraView(CameraView.MAIN, perspectiveCamera, orthographicCamera);
-    this.globalDragAdapter.makeCameraActive(perspectiveCamera);
-    this.mainCameraViewTracker.setCameras(perspectiveCamera, orthographicCamera);
-    this.snapGrid.addCamera(perspectiveCamera);
+  private void setCameras() {
+    SymmetricPerspectiveCamera mainCamera = sceneCameraImp.getSgCamera();
+    SymmetricPerspectiveCamera layoutCamera = layoutCameraImp.getSgCamera();
+    OrthographicCamera orthographicCamera = orthographicCameraImp.getSgCamera();
+    this.globalDragAdapter.addCameraView(CameraView.MAIN, mainCamera, layoutCamera, orthographicCamera);
+    this.globalDragAdapter.makeCameraActive(mainCamera);
+    this.mainCameraViewTracker.setCameras(mainCamera, layoutCamera, orthographicCamera);
+    this.snapGrid.addCamera(mainCamera);
+    snapGrid.addCamera(layoutCamera);
```

Comment:
you have a single snapGrid without a this here.  we should at least be consistent at the function level 😅

Reply:
I cannot explain this, but I can correct it. I am not found of the extra `this`s so I am cleaning this^H^H^H^H method, and stopping myself from doing so at a more global level.